### PR TITLE
Tractor config proto 68

### PR DESCRIPTION
**Pull request by ethanrublee**
Adds stub of tractor config, and a sample program to print json (but no persistence on disk).

Also, this fixes a units inconsistency between hub motor and the kinematic model, now that they both share the wheel_radius value.

### CODE REVIEW

**Comment by ethanrublee** [comment]
FYI this validated on the robot to my best ability.

File: protos/farm_ng_proto/tractor/v1/tractor.proto [hunk 1]
```diff
@@ -9,3 +12,17 @@ message TractorState {
...
+  google.protobuf.DoubleValue wheel_baseline = 1;
+  // Radius of wheel, in meters.
+  google.protobuf.DoubleValue wheel_radius = 2;
+  // Hub motor gear ratio
+  google.protobuf.DoubleValue hub_motor_gear_ratio = 3;
+  // Hub motor polls
+  google.protobuf.Int32Value hub_motor_poll_pairs = 4;
+
+  // Extrinsic relationship of sensors and motors.
+  PoseTree extrinsics = 5;
```
Comment (isherman) [inline protos/farm_ng_proto/tractor/v1/tractor.proto:27]: I worry a bit about losing the type-safety of the names of the nodes in the tree. We'll be relying on string matching. Something more explicit that represents the tractor's expected calibration frames might be better. Thoughts?

Comment (ethanrublee) [reply protos/farm_ng_proto/tractor/v1/tractor.proto:27]: I agree - need some way to enforce the poses.  However, having a more generic representation makes the library/debugging tools a bit more flexible/decoupled from the application.  In general its hard to enforce invariants in protobuf.

A config check could enforce expected frames/connections in the datastructure.

I think likely this field is consumed and updated by a generic calibration routine.

We could use introspection in code to enable parsing of field names to contruct a pose tree datastructure.

```protobuf
SE3Pose tractor_base_pose_tracking_camera_front = 1;
SE3Pose tractor_base_pose_tracking_camera_rear = 2;
```

But I think a simple function for ensuring a pose tree has the expected frames is sufficient.  And then we can more easily leverage library code that performs on the tree data-structure.

Comment (ethanrublee) [reply protos/farm_ng_proto/tractor/v1/tractor.proto:27]: This field isn't used in this pull request, so could remove it and consider this later when we're using it during calibration/visualization.

Comment (isherman) [reply protos/farm_ng_proto/tractor/v1/tractor.proto:27]: I see the point. Though i think you could argue the library code layer should provide PoseTree<->TractorPoseTree conversion, for libraries that consume generic PoseTrees. Then we have the guarantees, for example, that any tractor config file that gets saved to disk has all the necessary reference frames (without enforcing it by convention in library code). I guess I think of it kind of like modeling a database, where there are benefits to referential integrity in the DB (the SQL approach) vs. enforcing it all in application code (the schemaless NoSQL approach).

**Review by isherman — COMMENTED** [review]
This looks ready to merge to me, but will let @mmcgowen2015 give approval after testing on hardware

File: protos/farm_ng_proto/tractor/v1/tractor.proto (hunk 1 above)
Comment (ethanrublee) [reply protos/farm_ng_proto/tractor/v1/tractor.proto:27]: I'm open to this idea. I think for now, i'll remove extrinsics and then we can iterate on a type safe convention with the first usecase for the PoseTree (when we land some camera/base calibration).

**Review by mabitter — APPROVED** [review]
Sweet.

**Pull request merged**